broker-daemon/block-order-worker/index.js

```diff
@@ -528,14 +528,21 @@ class BlockOrderWorker extends EventEmitter {
         .then(() => fsm.removeAllListeners())
     })
 
-    // We are hooking into the reject lifecycle event of a fill state machine to trigger
-    // the failure of a blockorder
+    /**
+     * We are hooking into the reject lifecycle event of a fill state machine to trigger the failure of a blockorder.
+     * If the error comes back with an ORDER_NOT_PLACED, it means that the order the fillStateMachine
+     * attempted to fill was not in a state to be filled and we should rework the blockOrder
+     */
     fsm.once('reject', () => {
-      this.failBlockOrder(blockOrder.id, fsm.fill.error)
-        .catch(e => {
+      fsm.removeAllListeners()
+      if (fsm.isRelayerError()) {
```

Comment:
`isRelayerError` adds logic from the fsm into this reject handler, where instead its better to use `fsm.shouldRetry` or something like that